### PR TITLE
D8CORE-1823 Fix link widget when the user inputs <front>

### PR DESCRIPTION
**Pull request by pookmish**
# READY FOR REVIEW

# Summary
- Fix the form when a user enters `<front>` to display that if they go back to edit the item.

# Need Review By (Date)
- 4/29

# Urgency
- low

# Steps to Test
1. Checkout this branch
1. create a page with a card component
1. populate the link field with `<front>`
1. save and verify the url for the card goes to the home page
1. edit that page and edit the card you created
1. verify the link url says `<front>` and not `/<front>`

# See Also
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)

### CODE REVIEW

File: js/src/Components/Widgets/LinkWidget.js [hunk 1]
```diff
@@ -70,8 +70,7 @@ export const LinkWidget = ({fieldId, defaultValue, onFieldChange, settings}) =>
 
     if (matchedEntity && matchedEntity.length === 2) {
       uri = `entity:node/${matchedEntity[1]}`
-    }
-    else if (parsedUrl.protocol === null) {
```
Comment (sherakama) [inline js/src/Components/Widgets/LinkWidget.js]: I find the else statement on a new line is much more readable. Which eslint settings are we using and can we have else statements on their own?

File: js/src/Components/Widgets/LinkWidget.js [hunk 2]
```diff
@@ -96,9 +95,13 @@ export const LinkWidget = ({fieldId, defaultValue, onFieldChange, settings}) =>
 
       if (parsedUri.pathname === '/') {
         displayString = '<front>' + uri.substr(uri.indexOf('/') + 1);
+      } else if (parsedUri.pathname.indexOf('/<front>') >= 0) {
+        // If the user inputs `<front>#some-anchor`, the saved data on the entity is saved as
+        // `internal:/<front>#some-anchor` and the pathname is `/<front>#some-anchor`. We want
+        // to clean that up for users.
```
Comment (sherakama) [inline js/src/Components/Widgets/LinkWidget.js:103]: TIL. Thanks for the comment.

**Comment by sherakama** [comment]
Do you have tests working for this yet? This seems like a good case for one.

File: js/src/Components/Widgets/LinkWidget.js (hunk 1 above)
Comment (pookmish) [reply js/src/Components/Widgets/LinkWidget.js]: Yeah i think my phpstorm keeps flip flopping with that for some reason. i'll break it down

**Comment by pookmish** [comment]
I haven't been able to get pure javascript based tests. I could set up codeception tests but it would require alot of setup outside this ticket

File: js/src/Components/Widgets/LinkWidget.js (hunk 2 above)
Comment (pookmish) [reply js/src/Components/Widgets/LinkWidget.js:103]: I'm not 100% sure if that how regular fields handle the data, but its how the data is saved at this time. so :man_shrugging:

**Review by sherakama — APPROVED** [review]
Ah ok, so just the formatting thing then this is gtg.

**Review by sherakama — APPROVED** [review]

**Pull request merged**